File: task2p1.c
/* 
 * task2p1.c
 * Hannah Soria
 * 5/2/2024
 * showing c must allocate and deallocate memory
 */

#include <stdio.h>
#include <stdlib.h>

int main() {
    int *ptr = (int *)malloc(sizeof(int)); // allocate memory
    *ptr = 30;
    printf("the value of the pointer is %d\n", *ptr);
    free(ptr); // free the memory
    return 0;
}

File: task2p1.swift
/**
 * task1p1.swift
 * Hannah Soria
 * 5/2/2024
 * demonstration of swift allocating and deallocate memory manually
 */

import Foundation

func main() {
    let ptr = UnsafeMutablePointer<Int>.allocate(capacity: 1) // allocating memory
    ptr.pointee = 30
    print("the value of the pointer is \(ptr.pointee)")
    ptr.deallocate() // deallocating memory
}

main()